Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/App.hs

```diff
@@ -105,10 +112,29 @@ runDynamicOfferDriverApp' appCfg = do
 
         logInfo ("Runtime created. Starting server at port " <> show (appCfg.port))
         pure flowRt'
-    runSettings settings $ App.run (App.EnvR flowRt' appEnv)
+    let timeoutMiddleware = timeoutEvent flowRt appEnv (responseLBS status503 [] "") appCfg.timeoutSeconds
+    runSettings settings $ timeoutMiddleware (App.run (App.EnvR flowRt' appEnv))
 
 convertToHashMap :: Map.Map String Http.ManagerSettings -> HashMap.HashMap Text Http.ManagerSettings
 convertToHashMap = HashMap.fromList . map convert . Map.toList
   where
     convert (k, v) = (toText' k, v)
     toText' = T.pack
+
+timeoutEvent :: R.FlowRuntime -> AppEnv -> NWI.Response -> Int -> Middleware
+timeoutEvent flowRt appEnv timeoutResponse seconds app req respond = do
```

Comment:
can you make it a common function that is being used by both BAP and BPP?
can be added to `lib`